File: email-backup.java/src/main/java/demo/EmailBackup.java
package demo;

import javax.mail.MessagingException;
import java.io.IOException;
import java.util.Date;

final public class EmailBackup {

    public static void main(String[] args) {
        try {
            Flags flags = Configuration.parseFlags(args);
            if (flags != null) {
                String folderName = Names.getCurrentFolderName();
                String zipName = Names.getZipName(folderName, new Date());

                Timer timer1 = new Timer("Zipping");
                new Zipper(zipName, flags, folderName).zip();
                timer1.finish();

                Timer timer2 = new Timer("Sending");
                Sender.send(zipName, flags);
                timer2.finish();
            }
        } catch (IOException | MessagingException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/EmailBackup.kt
package demo

import java.io.IOException
import java.util.Date
import javax.mail.MessagingException

public object EmailBackup {

    @JvmStatic public fun main(args: Array<String>) {
        try {
            val flags = Configuration.parseFlags(*args)
            if (flags != null) {
                val folderName = Names.getCurrentFolderName()
                val zipName = Names.getZipName(folderName, Date())

                val timer1 = Timer("Zipping")
                Zipper(zipName, flags, folderName).zip()
                timer1.finish()

                val timer2 = Timer("Sending")
                Sender.send(zipName, flags)
                timer2.finish()
            }
        } catch (e: IOException) {
            println(e.getMessage())
        } catch (e: MessagingException) {
            println(e.getMessage())
        }
    }
}
